### ISSUE TITLE
Providing `null` for a nullable relation results in error

### ISSUE THREAD

**Issue by schickling** [prisma/prisma-client-js#614]
## Versions

- `"prisma2": "^2.0.0-alpha.942"`
- `"@prisma/client": "^2.0.0-alpha.942",`
- OSX
- Node v12.1.0

## Schema

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model User {
  id           Int     @id @default(autoincrement())
  // Global unique identifier!
  phoneNumber  String  @unique
  // Pre-Computed values for fast access.
  currentRisk  Float?
  groupRisk    Float?
  events       Event[] @relation("UserEvents")
  interactions Event[] @relation("UserInteractions")
}

model Event {
  id          Int       @id @default(autoincrement())
  user        User      @relation("UserEvents")
  timestamp   DateTime  @default(now())
  eventType   EventType
  interaction User?     @relation("UserInteractions")
  // Attachments, handled by event type implementation
  // might change
  data        String?
}

enum EventType {
  // Having travelled in a high risk area before
  HighRiskArea
  // Moving to another part of Germany.
  Location
  // Social interaction with another user
  Interaction
  // Developed Sympthoms
  Sympthoms
  // Diagnosed Sick
  DiagnosedSick
  // Marker inserted in past timeline when covid is diagnosed, to ease model calculation.
  MarkerSick 
}
```

When trying to run the following query the I received the error message below:

```ts
  async locationEvent(
    state: string,
    region: string,
    timestamp: Moment = moment(),
  ) {
    await this.client.event.create({
      data: {
        eventType: EventType.Location,
        timestamp: timestamp.toDate(),
        user: {
          connect: { id: this.user.id },
        },
        // Best encoding!
        data: JSON.stringify({ state, region }),
        interaction: null
        // interaction: null // commenting this out fixes the problem
      },
    })
  }
```

![image](https://user-images.githubusercontent.com/1567498/77907669-48c2ca00-728a-11ea-9cf4-78980acb31ca.png)

## Current workaround

Commenting out `interaction: null` in the query above fixes the problem.

## Notes

This problem might be related to https://github.com/prisma/prisma2/issues/2007

**Comment by dpetrick** [comment]
A fix is available on latest alpha.

**Comment by janpio** [comment]
Can you confirm this fix please @schickling? `2.0.0-beta.3` should include these changes.
